### PR TITLE
all[minor]: Replace all LC package scripts with `@langchain/scripts`

### PR DESCRIPTION
**Pull request by bracesproul**
### TODO
Fix `@langchain/community` adding index files to build artifacts.

Scripts will now be imported via `@langchain/scripts`.
Functions can be imported, and args passed the same as before.

`langchain` & `@langchain/community` will need to pass in a second arg (after the `pre` arg: `true`) so the extra functions for those two pkgs can be executed.

Since we have them in their own package now, we can write all the files as `.ts` (lets goooooooo!!)

cc @nfcampos I had to update `identify-secrets.ts` so it would make ts happy. Pretty much just added more type narrowing.

`./scripts` still exists, for this package.

I dropped the `tsconfig.cjs.json` file because it was causing the build to fail due to code not being avail in cjs:
```bash
src/move-cjs-to-dist.ts:6:40 - error TS1343: The 'import.meta' meta-property is only allowed when the '--module' option is 'es2020', 'es2022', 'esnext', 'system', 'node16', or 'nodenext'.

6   return resolve(dirname(fileURLToPath(import.meta.url)), relativePath);
                                         ~~~~~~~~~~~


Found 1 error in src/move-cjs-to-dist.ts:6
```

### CODE REVIEW

File: libs/langchain-scripts/package.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    "test:int": "NODE_OPTIONS=--experimental-vm-modules jest --testPathPattern=\\.int\\.test.ts --testTimeout 100000 --maxWorkers=50%",
+    "format": "prettier --write \"src\" \"scripts\"",
+    "format:check": "prettier --check \"src\" \"scripts\""
+  },
+  "author": "LangChain",
+  "license": "MIT",
+  "dependencies": {
+    "@langchain/core": "workspace:^",
+    "rollup": "^4.5.2",
+    "typescript": "<5.2.0"
```
Comment (bracesproul) [inline libs/langchain-scripts/package.json:39]: was forced to move these to hard deps

**Comment by jacoblee93** [comment]
I still see `tsconfig.cjs.json`?

File: langchain/scripts/move-cjs-to-dist.js [hunk 2]
```diff
@@ -1,38 +1,17 @@
-import { resolve, dirname, parse, format } from "node:path";
```
Comment (jacoblee93) [inline langchain/scripts/move-cjs-to-dist.js:1]: I was hoping to remove this and other scripts - can we not just build these via `package.json` commands?

**Comment by jacoblee93** [comment]
Was hoping to replace most `scripts/` files with `package.json` commands and like a `langchain_build.config.json` field or something - would that be possible?

File: langchain/scripts/move-cjs-to-dist.js (hunk 2 above)
Comment (bracesproul) [reply langchain/scripts/move-cjs-to-dist.js:1]: Ah yeah I think that should be possible for them. Let me think on it and I'll get around to it tmro

File: langchain-core/index.cjs [hunk 3]
```diff
@@ -0,0 +1 @@
+module.exports = require('./dist/index.cjs');
```
Comment (jacoblee93) [inline langchain-core/index.cjs:1]: Artifacts?

File: langchain-core/package.json [hunk 4]
```diff
@@ -440,9 +444,6 @@
     "utils/types.d.ts",
     "vectorstores.cjs",
     "vectorstores.js",
-    "vectorstores.d.ts",
-    "index.cjs",
```
Comment (jacoblee93) [inline langchain-core/package.json:444]: Something weird happened?

File: langchain/scripts/check-tree-shaking.js [hunk 5]
```diff
@@ -1,30 +1,7 @@
...
-
-  return entrypoints;
-}
-
-export function listExternals() {
-  return [
-    ...Object.keys(packageJson.dependencies),
-    ...Object.keys(packageJson.peerDependencies),
+checkTreeShaking({
+  extraInternals: [
```
Comment (jacoblee93) [inline langchain/scripts/check-tree-shaking.js:4]: This should be in the `config.js` file no?

File: langchain/scripts/create-entrypoints.js [hunk 6]
```diff
@@ -812,248 +811,14 @@ const extraImportMapEntries = [
...
-      .join(",\n") + ",\n];\n";
-  fs.writeFileSync(
-    `../${pkg}/${importConstantsPath}`,
-    "// Auto-generated by `scripts/create-entrypoints.js`. Do not edit manually.\n\nexport const optionalImportEntrypoints = [\n" +
-      contents
-  );
-};
-
-const command = process.argv[2];
+const absTsConfigPath = path.resolve(process.cwd(), "tsconfig.json");
```
Comment (jacoblee93) [inline langchain/scripts/create-entrypoints.js:814]: Same as above - should be a field in `config.js`?

File: langchain/scripts/move-cjs-to-dist.js [hunk 7]
```diff
@@ -1,38 +1,17 @@
...
-
-      await writeFile(abs(`${dest}/${renamed}`), rewritten, "utf8");
-    }
-  }
-}
-
-moveAndRename("../dist-cjs", "../dist").catch((err) => {
-  console.error(err);
-  process.exit(1);
+moveAndRename({
```
Comment (jacoblee93) [inline langchain/scripts/move-cjs-to-dist.js:13]: Should be possible via config fields

File: langchain/src/agents/react/prompt.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
-export const FORMAT_INSTRUCTIONS /* #__PURE__ */ = `Use the following format:
+export const FORMAT_INSTRUCTIONS = `Use the following format:
```
Comment (jacoblee93) [inline langchain/src/agents/react/prompt.ts:1]: This is no longer necessary?

File: langchain/src/evaluation/comparison/pairwise.ts [hunk 9]
```diff
@@ -20,7 +20,7 @@ import { PROMPT, PROMPT_WITH_REFERENCES } from "./prompt.js";
 import { ConstitutionalPrinciple } from "../../chains/index.js";
 import { Criteria, CriteriaLike } from "../criteria/criteria.js";
 
-const SUPPORTED_CRITERIA: Record<Criteria, string> = /* #__PURE__ */ {
+const SUPPORTED_CRITERIA: Record<Criteria, string> = {
```
Comment (jacoblee93) [inline langchain/src/evaluation/comparison/pairwise.ts:23]: Why remove this? Was it redundant to begin with?

File: libs/create-langchain-integration/template/scripts/check-tree-shaking.js [hunk 10]
```diff
@@ -1,80 +1,5 @@
...
-  if (failed) {
-    process.exit(1);
-  } else {
-    console.log("Tree shaking checks passed!");
-  }
-}
-
-checkTreeShaking();
+checkTreeShaking({
+  extraInternals: [/node\:/, /@langchain\/core\//],
```
Comment (jacoblee93) [inline libs/create-langchain-integration/template/scripts/check-tree-shaking.js:4]: Make this a config field

**Comment by jacoblee93** [comment]
Looking solid, see comments

**Pull request merged**